### PR TITLE
Split out atoms, components, and themes packages

### LINKED ISSUE #213: Split components package into components, atoms, and themes
@samrose3

> Explore splitting the components package from the atoms (perhaps a css or atoms package) and also maybe a themes package?
> components => {components, css, themes}
> Some pros:
> - Allow non-React uses to leverage atoms
> - Split concerns of atomic styles and React component
> - House themes in separately for easier code-splitting and not to be bundled with the atoms
> - Improve versioning perhaps… I can see the React components maybe updating a lot, but atoms and themes updating on a different cadence
> Cons:
> - More packages

@chloerice 

> Coming from a perspective of developing in  web , importing from a several packages feels normal and actually helpful in finding what I need (quilt packages and the symlinked packages built and used within the web repo itself are both really modular in this way).

### PR DESCRIPTION
**Pull request by aveline**
Resolves #213

### CODE REVIEW

File: packages/components/src/index.ts [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-export * from './utilities';
+export * from './Box';
+// export * from './Center';
+// export * from './Container';
+// export * from './Flex';
+// export * from './Grid';
+// export * from './Inline';
+// export * from './Link';
+// export * from './Stack';
+// export * from './Text';
```
Comment (kyledurand) [inline packages/components/src/index.ts:10]: Why are these commented out?

**Review by kyledurand — APPROVED** [review]

File: packages/components/src/index.ts (hunk 1 above)
Comment (aveline) [reply packages/components/src/index.ts:10]: Oh they shouldn't be. I think that was left in from when we were debugging. Thanks for catching that.

**Comment by samrose3** [comment]
Something I hadn't considered when suggesting this change is how it will impact the way consumers might normally interact with these utilities.

As an example, checkout `Layout.tsx`. We need to update it from 

```tsx
import {style, composeStyles} from '@vanilla-extract/css';
import {atoms, vars} from '@polaris/components';

export const root = composeStyles(
  atoms({
    marginY: 4,
  }),
  style({
    fontFamily: vars.fonts.system,
  }),
);
```

to this

```
import {style, composeStyles} from '@vanilla-extract/css';
import {atoms} from '@polaris/atoms';
import {vars} from '@polaris/themes';

export const root = composeStyles(
  atoms({
    marginY: 4,
  }),
  style({
    fontFamily: vars.fonts.system,
  }),
);
```

Does this seem overly confusing where to get which resources for styling as a consumer? Wdyt @aveline and @kyledurand?

We could re-export `vars`, and `atoms` in the `@polaris/components` package, but that doesn't seem like a good practice since it couples the packages more tightly.

This would also mean importing styles would/could look like this:

```
import "@polaris/atoms/styles"
import adminTheme from "@polaris/themes/admin"

export default () => (
  <ThemeProvider theme={adminTheme}>
    <Box>My cool layout</Box>
  </ThemeProvider>
);
```

**Comment by samrose3** [comment]
Also wanted to note, there may be an alternative way of enabling tree-shaking support for these separate concerns (atoms, components, themes) by using the the [sideEffects](url) property in the `packages/components/package.json`.

```json
"sideEffects": [
  "themes/**/*",
  "atoms/**/*",
  "*.css.ts"
],
```

My primary goal with splitting things up to so users have more control over what dependencies they need and don't need.

**Comment by alex-page** [comment]
I think these should be close to components. The import flow is rough for consumers or we are getting technical maintenance adding them to this project.

**Comment by samrose3** [comment]
@alex-page agreed. I think this was a good learning experiment. Going to close this issue and PR for now, but we can certainly reopen and explore again if we see the need

**Pull request closed without merging**